crm_compassion/models/weekly_demand.py

```diff
@@ -99,12 +106,23 @@ def _compute_demand_events(self):
             week.number_children_events = allocate
             week.resupply_events = resupply
 
+    @api.multi
+    def _compute_demand_sub(self):
+        """ Compute average of SUB since one year. """
+        start_date = datetime.today() - timedelta(weeks=52)
+        sub_sponsored = self.env['recurring.contract'].search_count([
+            ('origin_id.type', '=', 'sub'),
+            ('start_date', '>=', fields.Date.to_string(start_date))
+        ])
+        return float(sub_sponsored) / 52.0
```

Comment:
Instead of returning the result, you should assign it. This is how computed fields work:
```
average_sub = float(sub_sponsored) / 52.0
for week in self:
   week.number_sub_sponsorship = average_sub
```